dispatcherd/service/metrics.py

```diff
@@ -49,26 +46,126 @@ def collect(self) -> Generator[Metric, Any, Any]:
             yield m
 
 
+class CustomHttpServer:
+    def __init__(self, registry: CollectorRegistry):
+        self.registry = registry
+        self.server: Optional[asyncio.Server] = None
+
+    async def handle_request(self, reader: asyncio.StreamReader, writer: asyncio.StreamWriter) -> None:
+        addr = writer.get_extra_info('peername')
+        logger.info(f"Received connection from {addr}")
+
+        request_line = await reader.readline()
+        if not request_line:
+            logger.info(f"Received blank line from {addr}, closing")
+            writer.close()
+            await writer.wait_closed()
+            return
+
+        request_line_str = request_line.decode('utf-8').strip()
+        logger.info(f"Request: {request_line_str}")
+
+        # Parse the request line (simplified parsing)
+        try:
+            method, _, _ = request_line_str.split()
+        except ValueError:
+            logger.warning(f"Could not parse request line: {request_line_str}")
+            # Respond with 400 Bad Request for malformed request line
+            response_headers = "HTTP/1.1 400 Bad Request\r\nContent-Length: 0\r\nConnection: close\r\n\r\n"
+            writer.write(response_headers.encode('utf-8'))
+            await writer.drain()
+            writer.close()
+            await writer.wait_closed()
+            return
+
+        # Read headers (and ignore them for now)
+        while True:
+            header_line = await reader.readline()
+            if header_line == b'\r\n':
+                break
+
+        if method == 'GET':
+            try:
+                metrics_data = generate_latest(self.registry)
+                body = metrics_data.decode('utf-8')
+                status_line = "HTTP/1.1 200 OK"
+                content_type = "text/plain; version=0.0.4; charset=utf-8"
+            except Exception as e:
+                logger.error(f"Error generating metrics: {e}")
+                body = "Error generating metrics"
+                status_line = "HTTP/1.1 500 Internal Server Error"
+                content_type = "text/plain; charset=utf-8"
+        else:
+            # For any other path or method, respond with 404 Not Found
+            body = "Not Found"
+            status_line = "HTTP/1.1 404 Not Found"
+            content_type = "text/plain; charset=utf-8"
+
+        response_headers = (
+            f"{status_line}\r\n" f"Content-Type: {content_type}\r\n" f"Content-Length: {len(body.encode('utf-8'))}\r\n" "Connection: close\r\n\r\n"
+        )
+
+        response = response_headers + body
+
+        writer.write(response.encode('utf-8'))
```

Comment:
I wonder what happens if the client closes the connection around when this is called.